adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralBannerLoader.m

```diff
@@ -0,0 +1,159 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterMintegralBannerLoader.h"
+#import "GADMAdapterMintegralUtils.h"
+#import "GADMediationAdapterMintegralConstants.h"
+
+#include <stdatomic.h>
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKBanner/MTGBannerAdView.h>
+#import <MTGSDKBanner/MTGBannerAdViewDelegate.h>
+
+static CGSize const mintegralBannerAdSize320x50 = (CGSize){320, 50};
+static CGSize const mintegralBannerAdSize320x100 = (CGSize){320, 100};
+static CGSize const mintegralBannerAdSize300x250 = (CGSize){300, 250};
+static CGSize const mintegralBannerAdSize728x90 = (CGSize){728, 90};
+
+@interface GADMAdapterMintegralBannerLoader ()
+<GADMediationBannerAd,
+MTGBannerAdViewDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralBannerLoader{
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationBannerLoadCompletionHandler _adLoadCompletionHandler;
+    
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationBannerAdEventDelegate> _adEventDelegate;
+    
+    /// The Mintegral banner ad.
+    MTGBannerAdView *_bannerAdView;
+}
+
+- (void)loadBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
+                   completionHandler:
+(nonnull GADMediationBannerLoadCompletionHandler)completionHandler{
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationBannerLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationBannerAdEventDelegate>(
+                                                                      _Nullable id<GADMediationBannerAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationBannerAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+        
+    UIViewController *rootViewController = adConfiguration.topViewController;
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    if ([GADMAdapterMintegralUtils isStringEmpty:adUnitId] ||
+        [GADMAdapterMintegralUtils isStringEmpty:placementId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID or Placement ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
```

Comment:
You mentioned that placement ID is optional previously, is that still the case?

Reply:
no,please ignore previous comments, placement id is required.